web/docs/connectors/http.md

```diff
@@ -1,17 +1,20 @@
 # http
 
-Loads bytes via HTTP.
+Loads and saves bytes via HTTP.
 
 ## Synopsis
 
 ```
-http [<method>] <url> [<item>..]
+http [-f|--form] [-j|--json] [-v|--verbose] [<method>] <url> [<item>..]
 ```
 
 ## Description
 
 The `http` loader performs a HTTP request and returns the bytes of the HTTP
-response body.
+response body. Similarly, the `http` saver also performs a HTTP request with the
+request body being the the provided bytes. While saver and loader support the
+exact syntax, the saver only accepts data from its upstream operator and ignores
+any explicitly provided request body.
```

Comment:
We probably want to make it very clear that the incoming stream is sent incrementally, potentially over multiple requests. Also, is this behavior expected as a default by users?

Reply:
Ultimately we need both. The internal control knob we have is the `batch` operator. But we don't want to go there.

I think the correct UX is to provide a toggle that users can switch between these two options:

1. Buffer everything and create a single request body (default)
2. Use HTTP chunked encoding to send the data piecemeal (`--chunked`)

See `--chunked` from [HTTPie](https://httpie.io/docs/cli/redirected-input), which is our reference UX:

> By default, stdin data is buffered and then with no further processing used as the request body. If you provide Content-Length, then the request body is streamed without buffering. You may also use `--chunked` to enable streaming via [chunked transfer encoding](https://httpie.io/docs/cli/chunked-transfer-encoding#chunked-transfer-encoding) or `--compress`, `-x` to [compress the request body](https://httpie.io/docs/cli/chunked-transfer-encoding#compressed-request-body).

If there are no objections, I'd switch to buffering by default and add a `--chunked` option for streaming.

Reply:
We're sending one request per chunk now. For the scope of this already too large PR, I'd like to leave it as that. We can add more fancy batching later on. Users can already specify chunked transfer encoding via `--chunked`. I think we are on the same page that we have some issues with byte streams and how to reasonably stuff them in HTTP request bodies, but we can only fix that after your planned changes in the data model.

Resolving this for now.